main/lsp/LSPPathType.h

```diff
@@ -0,0 +1,23 @@
+
+#ifndef SORBET_LSP_PATH_TYPE_H
+#define SORBET_LSP_PATH_TYPE_H
+
+
+namespace sorbet::realmain::lsp {
+
+/**
+ * The enum declares a quality of service for a given change
+ */
+enum class PathType {
```

Comment:
Regarding naming I also was considering about `QualityOfService`, `TypecheckingQuality` and `TypecheckingPath`